### PR TITLE
ankama-launcher: 3.12.26 -> 3.12.27

### PR DESCRIPTION
**Pull request by Harbiinger**
## Things done
 
cc @GaetanLepage 

* Built on platform(s)
  * [x]  x86_64-linux
   * [ ]  aarch64-linux
   * [ ]  x86_64-darwin
   * [ ]  aarch64-darwin
* For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  * [ ]  `sandbox = relaxed`
  * [ ]  `sandbox = true`
* [ ]  Tested, as applicable:
  * [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  * and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  * or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  * made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
* [x]  Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
* [x]  Tested basic functionality of all binary files (usually in `./result/bin/`)
   * [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
    * [ ]  (Package updates) Added a release notes entry if the change is major or breaking
    * [ ]  (Module updates) Added a release notes entry if the change is significant
    * [ ]  (Module addition) Added a release notes entry if adding a new NixOS module
 * [x]  Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

**Note:** I have removed the text about the wine error since it was only occuring with Dofus 2 which is not available anymore.

 Add a 👍 [reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to [pull requests you find important](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

### CODE REVIEW

File: pkgs/by-name/an/ankama-launcher/package.nix [hunk 1]
```diff
@@ -48,9 +48,6 @@ appimageTools.wrapType2 {
     description = "Ankama Launcher";
     longDescription = ''
       Ankama Launcher is a portal that allows you to access Ankama's video games, VOD animations, webtoons, and livestreams, as well as download updates, stay up to date with the latest news, and chat with your friends.
-
-      If you encounter a `wine` error while running *Dofus*, delete or rename the `cinematics/` directory:
-      - Go to the directory where you installed the game and run: `mv content/gfx/cinematics content/gfx/cinematics_DISABLE`
```
[the remark this answers is not in the document]
Comment (Harbiinger) [reply pkgs/by-name/an/ankama-launcher/package.nix:53]: It was only with Dofus 2. I put a note at the end of the pr's description

[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/by-name/an/ankama-launcher/package.nix:53]: Oh right. Thanks !

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363713`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ankama-launcher</li>
  </ul>
</details>

---
### `aarch64-linux`

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Review by GaetanLepage — APPROVED** [review]
LGTM thanks !

**Comment by nezia1** [comment]
Hey, I've ran into an issue with your package (I actually ran in the same issue myself trying to package it.). It seems that whenever the launcher self updates, it completely stops working, making the launcher unusable despite having deleted `$XDG_CONFIG_HOME/zaap`. These are the logs I get:
```
12/10/2024, 8:53:56 AM.5 - info: [AUTO_UPDATER] Has checked for updates true
12/10/2024, 8:53:56 AM.5 - info: [AUTO_UPDATER] An update is available
12/10/2024, 8:53:56 AM.9 - error: [1015 AUTO_UPDATER] error, Error: APPIMAGE env is not defined
    at Object.t.newError (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:1:54148)
    at Object.task (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:184:188162)
    at d.executeDownload (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:71:80166) Error: APPIMAGE env is not defined
    at Object.t.newError (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:1:54148)
    at Object.task (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:184:188162)
    at d.executeDownload (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:71:80166)
12/10/2024, 8:53:56 AM.9 - error: [MAIN] Autoupdater failed
12/10/2024, 8:53:56 AM.9 - info: [MAIN] primarySetups
12/10/2024, 8:53:56 AM.11 - error: [1074 MAIN] A fatal error occurred (unhandled rejection) Error: APPIMAGE env is not defined
    at Object.t.newError (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:1:54148)
    at Object.task (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:184:188162)
    at d.executeDownload (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:71:80166)
(node:171092) UnhandledPromiseRejectionWarning: Error: APPIMAGE env is not defined
    at Object.t.newError (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:1:54148)
    at Object.task (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:184:188162)
    at d.executeDownload (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:71:80166)
(Use `zaap --trace-warnings ...` to show where the warning was created)
(node:171092) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
12/10/2024, 8:53:56 AM.22 - error: [1083 SENTRY] Sending error data to sentry, event id: bd182904b5334d68a6837c7a936e90ce
12/10/2024, 8:53:56 AM.30 - info: [I18next] Successfully initialized
12/10/2024, 8:53:56 AM.30 - info: [UNIVERSE] setup
12/10/2024, 8:53:56 AM.122 - info: [UNIVERSE] Downloaded universes config
12/10/2024, 8:53:57 AM.374 - info: No settings for service games in universe /home/nezia/.config/zaap/universe/Krosmoz
12/10/2024, 8:53:57 AM.382 - info: [REGISTRY] setup
12/10/2024, 8:53:57 AM.384 - debug: [REGISTRY] listening repository update
12/10/2024, 8:53:57 AM.386 - info: No settings for service stream in universe /home/nezia/.config/zaap/universe/Krosmoz
12/10/2024, 8:53:57 AM.386 - info: No settings for service webtoon in universe /home/nezia/.config/zaap/universe/Krosmoz
12/10/2024, 8:53:57 AM.386 - info: No settings for service shop in universe /home/nezia/.config/zaap/universe/Krosmoz
12/10/2024, 8:53:57 AM.386 - warn: [REGISTRY] Failed to load local repository file: Error: ENOENT: no such file or directory, open '/home/nezia/.config/zaap/repositories/repositories.json'
12/10/2024, 8:53:57 AM.391 - info: No settings for service video in universe /home/nezia/.config/zaap/universe/Krosmoz
12/10/2024, 8:53:57 AM.641 - debug: [REGISTRY] update
12/10/2024, 8:53:57 AM.644 - warn: [UNIVERSE] Repository was set but could not fetch initial games list TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'ClientRequest'
    |     property 'res' -> object with constructor 'IncomingMessage'
    --- property 'req' closes the circle
    at JSON.stringify (<anonymous>)
    at e.exports.saveRepositoryListFile (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:459:162939)
    at e.exports.update (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:459:160346)
    at /nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:459:163039
12/10/2024, 8:53:57 AM.684 - debug: [REGISTRY] update
12/10/2024, 8:53:57 AM.685 - debug: [REGISTRY] Parse repository Data
12/10/2024, 8:53:57 AM.690 - warn: [INFORMATION] error while checking dataDir
12/10/2024, 8:53:57 AM.690 - warn: [INFORMATION] error while checking dataDir
12/10/2024, 8:53:57 AM.690 - warn: [INFORMATION] error while checking dataDir
12/10/2024, 8:53:57 AM.691 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:57 AM.691 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:57 AM.691 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:57 AM.691 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:57 AM.692 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:57 AM.692 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:57 AM.702 - warn: [INFORMATION] error while checking dataDir
12/10/2024, 8:53:57 AM.704 - warn: [INFORMATION] error while checking dataDir
12/10/2024, 8:53:57 AM.706 - warn: [INFORMATION] error while checking dataDir
12/10/2024, 8:53:57 AM.707 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:57 AM.707 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:57 AM.707 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:57 AM.707 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:57 AM.707 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:57 AM.708 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:57 AM.728 - warn: [INFORMATION] error while checking dataDir
12/10/2024, 8:53:57 AM.729 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:57 AM.729 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:57 AM.917 - info: No settings for service webtoon in universe /home/nezia/.config/zaap/universe/Allskreen
12/10/2024, 8:53:57 AM.918 - info: No settings for service shop in universe /home/nezia/.config/zaap/universe/Allskreen
12/10/2024, 8:53:58 AM.177 - warn: [ZB-514] setConfiguration (load)
12/10/2024, 8:53:58 AM.215 - debug: [GAME_CONFIG] New configuration dofus dofus3
12/10/2024, 8:53:58 AM.252 - warn: [ZB-514] setConfiguration (load)
12/10/2024, 8:53:58 AM.252 - warn: [ZB-514] setConfiguration (load)
12/10/2024, 8:53:58 AM.253 - warn: [ZB-514] setConfiguration (load)
12/10/2024, 8:53:58 AM.255 - warn: [ZB-514] setConfiguration (load)
12/10/2024, 8:53:58 AM.255 - warn: [ZB-514] setConfiguration (load)
12/10/2024, 8:53:58 AM.271 - debug: [GAME_CONFIG] New configuration retro main
12/10/2024, 8:53:58 AM.296 - debug: [GAME_CONFIG] New configuration dofus main
12/10/2024, 8:53:58 AM.322 - debug: [GAME_CONFIG] New configuration wakfu main
12/10/2024, 8:53:58 AM.339 - debug: [GAME_CONFIG] New configuration dofus beta
12/10/2024, 8:53:58 AM.359 - debug: [GAME_CONFIG] New configuration wakfu beta
12/10/2024, 8:53:58 AM.466 - warn: [ZB-514] setConfiguration (load)
12/10/2024, 8:53:58 AM.478 - debug: [GAME_CONFIG] New configuration retro beta
12/10/2024, 8:53:58 AM.646 - info: [REGISTRY] setup
12/10/2024, 8:53:58 AM.647 - debug: [REGISTRY] listening repository update
12/10/2024, 8:53:58 AM.648 - debug: [REGISTRY] Parse repository Data
12/10/2024, 8:53:58 AM.652 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:58 AM.653 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:58 AM.654 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:58 AM.654 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:58 AM.663 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:58 AM.663 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:58 AM.666 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:58 AM.666 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:58 AM.669 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:58 AM.671 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:53:58 AM.675 - debug: [1121 INFORMATION] Failed to load information data (information.json)
12/10/2024, 8:53:58 AM.676 - debug: [1121 INFORMATION] Failed to load game file release.json
12/10/2024, 8:54:05 AM.521 - info: [UNIVERSE] Register universe Krosmoz
12/10/2024, 8:54:05 AM.521 - info: [UNIVERSE] Register universe Allskreen
12/10/2024, 8:54:05 AM.522 - info: [UNIVERSE] krosmoz selected
12/10/2024, 8:54:05 AM.552 - error: [AppError] 
{
  "code": 8001,
  "message": "No API KEY found on machine",
  "displayMessageOnFront": true,
  "complement": {},
  "name": "Error",
  "stack": "Error: No API KEY found on machine\n    at u.authenticateFromStoredApiKey (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:1:64859)\n    at u.setup (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:1:57303)\n    at i.primarySetups (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:71:99315)\n    at i.bootstrapOnce (/nix/store/0zvga75q8r15dydn77fvg291f63aij9x-ankama-launcher-3.12.24-extracted/resources/app.asar/main.js:71:98545)"
}
12/10/2024, 8:54:05 AM.559 - info: [SHOP] setup
12/10/2024, 8:54:05 AM.562 - info: [PIP] setup
12/10/2024, 8:54:05 AM.563 - info: [OVERLAY] setup
12/10/2024, 8:54:05 AM.565 - info: [AUTO_LAUNCH] setting to false
12/10/2024, 8:54:05 AM.566 - info: [NEWS] setup
12/10/2024, 8:54:05 AM.567 - info: [CHANGELOGS] setup
12/10/2024, 8:54:05 AM.567 - info: [TERMS] setup
12/10/2024, 8:54:05 AM.575 - debug: [REGISTRY] update
12/10/2024, 8:54:05 AM.590 - debug: [REGISTRY] Parse repository Data
```

As we can see above, it looks like it's related to the APPIMAGE env not being defined somehow, and some of the program's logic relying on that to self-update. There's also this API_KEY issue, which I'm unsure if it's even related.

**Pull request merged**